controllers/cronset_controller.go

```diff
@@ -57,15 +57,34 @@ func (r *CronSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&corev1.Node{}).
 		Watches(&source.Kind{Type: &corev1.Node{}},
 			handler.EnqueueRequestsFromMapFunc(func(node client.Object) []reconcile.Request {
+				nodeLabels := node.GetLabels()
+				r.Log.Info("Node Event", "Node", node.GetName(), "Node Labels", nodeLabels)
+
 				var cronSetObjs batchv1alpha1.CronSetList
 				_ = mgr.GetClient().List(context.TODO(), &cronSetObjs)
 
 				var requests []reconcile.Request
-				for _, obj := range cronSetObjs.Items {
+
+			CronSetLoop:
+				for _, cronSet := range cronSetObjs.Items {
+					nodeSelector := cronSet.Spec.CronJobTemplate.Spec.JobTemplate.Spec.Template.Spec.NodeSelector
+					r.Log.Info("Check CronSet for Node Event", "CronSet name", cronSet.Name, "nodeSelector", nodeSelector)
+
+					if len(nodeSelector) != 0 {
+						for key, value := range nodeSelector {
+							nodeLabel, exists := nodeLabels[key]
+							if !exists || nodeLabel != value {
+								r.Log.Info("CronSet's nodeSelector doesn't match event trigger node's label", "Node", node.GetName(), "CronSet", cronSet.Name)
+								continue CronSetLoop
```

Comment:
nit: Since there is no more work to be done after this, it seems like it would have the same effect as using `break` alone. Is there any other reason why `label continue` is used?

Reply:
I used break pattern at first. but that way needs one variable to break outer for loop(cronSetLoop). the variable is the boolean type as matching label or not.

As you know, every case of not match label, the request must not be added. so I have to add the if condition with the variable so that it skips adding request and continues the outer loop.

so I judged label is better than the break pattern to the code from a refactoring point.